crypto/dilithium/p_pqdsa_test.cc

```diff
@@ -373,65 +371,61 @@ TEST_P(PQDSAParameterTest, KAT) {
     ASSERT_TRUE(t->GetBytes(&sk, "sk"));
-    ASSERT_TRUE(t->GetAttribute(&smlen, "smlen"));
+    ASSERT_TRUE(t->GetBytes(&msg, "msg"));
+    ASSERT_TRUE(t->GetAttribute(&mlen, "mlen"));
     ASSERT_TRUE(t->GetBytes(&sm, "sm"));
+    ASSERT_TRUE(t->GetAttribute(&smlen, "smlen"));
+    ASSERT_TRUE(t->GetBytes(&ctxstr, "ctx"));
 
     size_t pk_len = GetParam().public_key_len;
     size_t sk_len = GetParam().private_key_len;
     size_t sig_len = GetParam().signature_len;
+    std::string name = GetParam().name;
 
-    // The KAT files generated by the dilithium team use the optional APIs that
-    // create a signature for a message m and append the message to the end of
-    // the signature. We only want to bring the APIs that create and verify just
-    // the signature, therefore each signature is a constant
-    // DILITHIUM3_SIGNATURE_BYTES and we truncate the KAT's signed message down,
-    // "sm" down to a constant DILITHIUM3_SIGNATURE_BYTES.
-
+    std::vector<uint8_t> pub(pk_len);
+    std::vector<uint8_t> priv(sk_len);
     std::vector<uint8_t> signature(sig_len);
-    sm.resize(sig_len);
 
-    // Convert string read from KAT to int
     size_t mlen_int = std::stoi(mlen);
+    sm.resize(sig_len);
 
-    // Here we fix the DRBG (AES-CTR) so that we are able to seed it with the
-    // seed from the KAT (testing only)
-    pq_custom_randombytes_use_deterministic_for_testing();
-    pq_custom_randombytes_init_for_testing(seed.data());
-
-    // Generate our dilithium public and private key pair
-    bssl::UniquePtr<EVP_PKEY_CTX> pctx(EVP_PKEY_CTX_new_id(EVP_PKEY_PQDSA, nullptr));
-    ASSERT_TRUE(pctx);
-    ASSERT_TRUE(EVP_PKEY_CTX_pqdsa_set_params(pctx.get(),GetParam().nid));
-    ASSERT_TRUE(EVP_PKEY_keygen_init(pctx.get()));
-    EVP_PKEY *raw = nullptr;
-    ASSERT_TRUE(EVP_PKEY_keygen(pctx.get(), &raw));
-    bssl::UniquePtr<EVP_PKEY> pkey(raw);
-
-    // Compare the expected public/secret key from KATs with generated values
-    CMP_VEC_AND_PKEY_PUBLIC(pk, pkey, pk_len);
-    CMP_VEC_AND_PKEY_SECRET(sk, pkey, sk_len);
-
-    // Generate a signature for the message
-    // We use EVP_DigestSign because dilithium supports the use of
-    // non-hash-then-sign (just like ed25519) so we first init EVP_DigestSign
-    // WITHOUT a hash function.
-    bssl::ScopedEVP_MD_CTX ctx;
-    ASSERT_TRUE(EVP_DigestSignInit(ctx.get(), nullptr, nullptr, nullptr, pkey.get()));
-    ASSERT_TRUE(EVP_DigestSign(ctx.get(), signature.data(), &sig_len, msg.data(), mlen_int));
-    EXPECT_EQ(Bytes(sm), Bytes(signature.data(), sig_len));
-    ctx.Reset();
-
-    // Verify the signature
-    ASSERT_TRUE(EVP_DigestVerifyInit(ctx.get(), nullptr, nullptr, nullptr, pkey.get()));
-    ASSERT_TRUE(EVP_DigestVerify(ctx.get(), signature.data(), sig_len, msg.data(), mlen_int));
+    // Generate key pair from seed xi and assert that public and private keys
+    // are equal to expected values from KAT
+    if (name == "MLDSA65") {
```

Comment:
added a lookup/find in [061ebce](https://github.com/aws/aws-lc/pull/1999/commits/061ebceca90a012a0e8ba261f71c8724251a42ea)

Reply:
For reference, I put the check outside the FileTestGTest main loop so that we don't perform this check every iteration of the test vector loop. I tested functionality by renaming a test vector name.